packages/expo-apple-authentication/ios/EXAppleAuthentication/EXAppleAuthenticationRequest.m

```diff
@@ -0,0 +1,100 @@
+
+#import <EXAppleAuthentication/EXAppleAuthenticationRequest.h>
+#import <EXAppleAuthentication/EXAppleAuthenticationMappings.h>
+
+#import <UMCore/UMDefines.h>
+
+@interface EXAppleAuthenticationRequest ()
+
+@property (nonatomic, strong, nonnull) NSDictionary *options;
+@property (nonatomic, strong, nullable) void(^callback)(NSDictionary *, NSError *);
+@property (nonatomic, strong, nonnull) ASAuthorizationController *authController;
+
+@end
+
+@implementation EXAppleAuthenticationRequest
+
+- (instancetype)initWithOptions:(NSDictionary *)options
+                    andCallback:(void(^)(NSDictionary *response, NSError *error))callback
+{
+  if (self = [super init]) {
+    _options = options;
+    _callback = callback;
+  }
+  return self;
+}
+
+- (void)performOperation:(ASAuthorizationProviderAuthorizationOperation)operation
+{
+  ASAuthorizationAppleIDProvider* appleIDProvider = [[ASAuthorizationAppleIDProvider alloc] init];
+  ASAuthorizationAppleIDRequest* request = [appleIDProvider createRequest];
+
+  NSArray<NSNumber *> *requestedScopes = _options[@"requestedScopes"];
+
+  request.requestedScopes = [EXAppleAuthenticationMappings importScopes:requestedScopes];
+  request.requestedOperation = operation;
+
+  if (_options[@"user"]) {
+    request.user = _options[@"user"];
+  }
+  if (_options[@"state"]) {
+    request.state = _options[@"state"];
+  }
+
+  _authController = [[ASAuthorizationController alloc] initWithAuthorizationRequests:@[request]];
+  _authController.presentationContextProvider = self;
+  _authController.delegate = self;
+
+  [_authController performRequests];
+}
+
++ (EXAppleAuthenticationRequest *)performOperation:(ASAuthorizationProviderAuthorizationOperation)operation
+                                       withOptions:(NSDictionary *)options
+                                      withCallback:(void(^)(NSDictionary *, NSError *))callback
+{
+  EXAppleAuthenticationRequest *authRequest = [[EXAppleAuthenticationRequest alloc] initWithOptions:options andCallback:callback];
+  [authRequest performOperation:operation];
+  return authRequest;
+}
+
+#pragma mark - ASAuthorizationControllerDelegate
+
+- (void)authorizationController:(ASAuthorizationController *)controller
+   didCompleteWithAuthorization:(ASAuthorization *)authorization
+{
+  ASAuthorizationAppleIDCredential *credential = authorization.credential;
+  NSDictionary *response = @{
+                         @"fullName": [EXAppleAuthenticationMappings exportFullName:credential.fullName],
+                         @"email": UMNullIfNil(credential.email),
+                         @"user": credential.user,
+                         @"authorizedScopes": credential.authorizedScopes,
+                         @"realUserStatus": @(credential.realUserStatus),
+                         @"state": UMNullIfNil(credential.state),
+                         @"authorizationCode": [EXAppleAuthenticationMappings exportData:credential.authorizationCode],
```

Comment:
credential.authorizationCode and credential.identityToken are nullable -- need to handle this. If they never should be null, we should throw (can implement this assertion in the API layer and allow the platform layer's interface to declare these two fields as nullable).